test/spec_request.rb

```diff
@@ -114,32 +114,38 @@ class RackRequestTest < Minitest::Spec
     req = make_request \
       Rack::MockRequest.env_for("/", "HTTP_HOST" => "www2.example.org")
     req.host.must_equal "www2.example.org"
+    req.hostname.must_equal "www2.example.org"
 
     req = make_request \
       Rack::MockRequest.env_for("/", "SERVER_NAME" => "example.org", "SERVER_PORT" => "9292")
     req.host.must_equal "example.org"
+    req.hostname.must_equal "example.org"
 
     req = make_request \
       Rack::MockRequest.env_for("/", "HTTP_HOST" => "localhost:81", "HTTP_X_FORWARDED_HOST" => "example.org:9292")
     req.host.must_equal "example.org"
+    req.hostname.must_equal "example.org"
 
     req = make_request \
       Rack::MockRequest.env_for("/", "HTTP_HOST" => "localhost:81", "HTTP_X_FORWARDED_HOST" => "[2001:db8:cafe::17]:47011")
-    req.host.must_equal "2001:db8:cafe::17"
+    req.host.must_equal "[2001:db8:cafe::17]"
+    req.hostname.must_equal "2001:db8:cafe::17"
 
     req = make_request \
       Rack::MockRequest.env_for("/", "HTTP_HOST" => "localhost:81", "HTTP_X_FORWARDED_HOST" => "2001:db8:cafe::17")
-    req.host.must_equal "2001:db8:cafe::17"
+    req.host.must_equal "[2001:db8:cafe::17]"
+    req.hostname.must_equal "2001:db8:cafe::17"
 
     env = Rack::MockRequest.env_for("/", "SERVER_ADDR" => "192.168.1.1", "SERVER_PORT" => "9292")
     env.delete("SERVER_NAME")
     req = make_request(env)
     req.host.must_equal "192.168.1.1"
+    req.hostname.must_equal "192.168.1.1"
 
     env = Rack::MockRequest.env_for("/")
     env.delete("SERVER_NAME")
     req = make_request(env)
-    req.host.must_equal ""
+    req.host.must_be_nil
```

Comment:
I think removing this test completely is better.  `SERVER_NAME` is required by SPEC.  A Request without `SERVER_NAME` should be undefined behavior.

Reply:
It's an existing spec. We could remove it. I wonder why lint doesn't fail?